exercise/api/views.py

```diff
@@ -424,6 +428,39 @@ def grader_detail(self, request, *args, **kwargs):
 
         return Response(_post_async_submission(request, self.submission.exercise, self.submission))
 
+    @action(
+        detail=True,
+        url_path='re-submit',
+        url_name='re-submit',
+        get_permissions = lambda: [OnlyCourseStaffPermission()],
+        methods=['get'],
+    )
+    def resubmit(self, request, *args, **kwargs):
+        if not self.submission.exercise.is_submittable:
+            return self.http_method_not_allowed(request, *args, **kwargs)
+
+        data = None
+
+        page = self.submission.exercise.grade(self.submission, request)
+
+        # Enroll after succesfully resubmitting to the enrollment exercise.
+        if self.submission.exercise.status in (
+            LearningObject.STATUS.ENROLLMENT,
+            LearningObject.STATUS.ENROLLMENT_EXTERNAL,
+        ) and self.submission.status == Submission.STATUS.READY:
+            self.instance.enroll_student(request.user)
```

Comment:
Why would we need to check enrollment questionnaires in regrading? Well, it could be possible to regrade a somehow temporarily broken submission in the enrollment questionnaire. What was your idea behind this?

Reply:
I was thinking exactly that, if it's somehow possible to have an enrollment questionnaire stuck in grading, then it could be resubmitted. But is it correct that only submit exercises can get stuck in grading?